SEN-2555 Add migrations for JUnit fail

seanf opened the pull request:
https://securecodewarrior.atlassian.net/browse/SEN-2555

bbogason reviewed (approved):
Same comment as in other PR: I only reviewed this one commit: b50c0a0545f97caf9950d3502eeebfb249ebcd22

I would prefer if each PR only contained the relevant commits for that ticket.

seanf commented on example_code/assertj/src/test/java/junit/JUnit5Assertions.java:
```diff
@@ -273,4 +274,58 @@ void assertDoesNotThrow() {
 
+    @SuppressWarnings({"TrivialFunctionalExpressionUsage", "RedundantSuppression"})
+    @Test
+    void testFail() {
```
Sorry, I confused two different 'fail' tickets. This work should have been marked as SEN-2555. 

The idiomatic case you have identified seems like something which may not be possible with Sensei right now, a bit like https://securecodewarrior.atlassian.net/browse/SEN-2396 and the feature request https://securecodewarrior.atlassian.net/browse/SEN-2406

seanf commented on example_code/assertj/src/test/java/junit/JUnit5Assertions.java:
```diff
@@ -273,4 +274,58 @@ void assertDoesNotThrow() {
+        Assertions.fail("message", new Throwable());
+        Assertions.fail(new Throwable());
+
+        // should change to these calls respectively:
```
Sorry, I imagine it's harder to read in a PR, but I find this way much, much easier to compare in the IDE. I go to the marked code above, use Alt-Enter a few times to "fix all" and then compare the "fixed" code above this comment with the expected code below. I find it makes it much easier to make visual comparisons for large numbers of assertions, without lots of visual noise.

Maybe for future examples we should try a Before.java and Expected.java, so that they can be put side by side in the IDE (split mode). Or would that be even worse for GitHub reviews?

seanf commented:
> Same comment as in other PR: I only reviewed this one commit: b50c0a0

Now rewritten as 18757d855d38e739ba92eb6f8cb29273dd19df3f and rebased on top of the merged https://github.com/SecureCodeWarrior/sensei-recipes/pull/33
